spyder/plugins/editor/plugin.py

```diff
@@ -2316,18 +2313,17 @@ def add_cursor_position_to_history(self, filename, position, fc=False):
         self.cursor_pos_index = len(self.cursor_pos_history)-1
         self.update_cursorpos_actions()
 
-    def cursor_moved(self, filename0, position0, filename1, position1):
-        """Cursor was just moved: 'go to'"""
-        if position0 is not None:
-            self.add_cursor_position_to_history(filename0, position0)
-        self.add_cursor_position_to_history(filename1, position1)
 
     def text_changed_at(self, filename, position):
         self.last_edit_cursor_pos = (to_text_string(filename), position)
 
     def current_file_changed(self, filename, position):
         self.add_cursor_position_to_history(to_text_string(filename), position,
                                             fc=True)
+        code_editor = self.get_current_editor()
+        code_editor.sig_cursor_position_changed.connect(
+            self.current_editor_cursor_changed)

```

Comment:
@steff456, there's no need to do this here because:

* `sig_cursor_position_changed` is connected to `editor_cursor_position_changed` here:

    https://github.com/spyder-ide/spyder/blob/d1adacfc7fecf529bb49bc0cd511925300825f97/spyder/plugins/editor/widgets/editor.py#L2590-L2591

* That method in turn emits another signal called `sig_editor_cursor_position_changed`:

    https://github.com/spyder-ide/spyder/blob/d1adacfc7fecf529bb49bc0cd511925300825f97/spyder/plugins/editor/widgets/editor.py#L2628-L2630

* The editor plugin can connect to that signal and change the cursor position history accordingly. For example, here it connects to that signal to update the status bar widget that shows the current line and column:

    https://github.com/spyder-ide/spyder/blob/d1adacfc7fecf529bb49bc0cd511925300825f97/spyder/plugins/editor/plugin.py#L1261-L1262

----

So instead of connecting directly to `sig_cursor_position_changed`, please connect to `sig_editor_cursor_position_changed` (below line 1262) like this:

```python
editorstack.sig_editor_cursor_position_changed.connect(
   self.current_editor_cursor_changed)
```